### ISSUE TITLE
Figure out why netty EpollEventLoop isn't working on Lambda

### ISSUE THREAD

**Issue by kiiadi** [aws/aws-sdk-java-v2#193]
Refer to https://gitter.im/aws/aws-sdk-java-v2 conversation between @kiiadi & @wjbuys 
https://gist.github.com/wjbuys/7634406ba09ad0ae4f0564db948077ef#file-lambda-stacktrace-log

**Comment by millems** [comment]
I've not been able to reproduce this, so it may be a bug in the SDK that has since been fixed.

**Comment by millems** [comment]
Marking this as a feature request: Support EpollEventLoop on Lambda.

**Comment by kiiadi** [comment]
Yeah we used to have a gate that explicitly disabled `Epoll` on Lambda - but now it seems like we just use the `NioEventLoopGroup` for everything : https://github.com/aws/aws-sdk-java-v2/blob/cdfb7d3f6a7dafb37be84f6e95e49feb0d0af273/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/SdkEventLoopGroup.java#L136

Do we use Epoll at all ? Is the request to support it on Lambda or just support it more generally.

**Comment by dagnir** [comment]
@kiiadi Yes that's right we don't use available native bindings at the moment. We change this to enable those binds when/where we can.

**Comment by WorkGitH** [comment]
I am curious on the state of this issue. I see that the initial issue is with lambda, however, I am confused on why EpollEventLoop was disable for every use case. Is there a recommend way to utilize the EpollEventLoop in the NettyHttpClient?

**Comment by samarprakash** [comment]
We can create our EpollEventLoopGroup and inject it to SdkAsyncHttpClient as below.

public EventLoopGroup awsSdkEpollEventLoopGroup() {
        return Epoll.isAvailable()
                ? new EpollEventLoopGroup(defaultWorkerThreadCount, threadFactory)
                : new NioEventLoopGroup(defaultWorkerThreadCount, threadFactory);
}

public SdkEventLoopGroup sdkEventLoopGroup() {
        return SdkEventLoopGroup.create(this.awsSdkEpollEventLoopGroup, EpollSocketChannel::new);
}

public SdkAsyncHttpClient sdkAsyncHttpClient(final SdkEventLoopGroup sdkEventLoopGroup) {
        return NettyNioAsyncHttpClient.builder()
                .maxConcurrency(getIntProperty(AppProperties.AWS_ASYNC_CLIENT_MAX_HTTP_CONNECTIONS))
                .eventLoopGroup(sdkEventLoopGroup)
                .build();
}

**Comment by Hexcles** [comment]
It's been seven years since https://github.com/aws/aws-sdk-java-v2/pull/253/files and I can't find any more info on exactly what retry problems you ran into. It'd be great if we can at least have some brief description of the problem (and even better, an easy toggle to turn on epoll) so we can verify this at our own risk.
